### PR TITLE
Integration of BankAccountEditor from supplier

### PR DESCRIPTION
**Pull request by PunkSage**

### CODE REVIEW

File: src/client/components/SupplierApplicationForm/SupplierApplicationForm.js [hunk 1]
```diff
@@ -53,7 +52,14 @@ class SupplierApplicationForm extends React.Component {
       jsFileName: 'contact-bundle'
     });
 
-    this.externalComponents = { SupplierEditor, SupplierAddressEditor, SupplierContactEditor };
+    const SupplierBankAccountEditor = serviceComponent({
+      serviceRegistry,
+      serviceName: 'supplier' ,
+      moduleName: 'supplier-bank_accounts',
+      jsFileName: 'bank_accounts-bundle'
```
Comment (gr4per) [inline src/client/components/SupplierApplicationForm/SupplierApplicationForm.js]: i dont see reason why jsFileName is different from module name?

File: src/client/components/SupplierApplicationForm/SupplierApplicationForm.js [hunk 2]
```diff
@@ -2,22 +2,20 @@
...
 import I18nBundle from '../Widgets/components/I18nBundle';
-import OnboardingUserService from '../../service/OnboardingUserService';
 import serviceComponent from '@opuscapita/react-loaders/lib/serviceComponent';
 
 class SupplierApplicationForm extends React.Component {
 
   static propTypes = {
     currentUserData: React.PropTypes.object
-  }
+  };
```
Comment (gr4per) [inline src/client/components/SupplierApplicationForm/SupplierApplicationForm.js]: i see plenty of non-changes by formatting changes or things like adding a colon like here.
this is actually making reviewing harder and i would request you to not change formatting unless you really change code.
u can easily run diff locally before commit and make sure you dont inlcude non changes.
possibly the editor also is auto changing things, but that could be turned off

File: src/client/components/SupplierApplicationForm/SupplierApplicationForm.js [hunk 3]
```diff
@@ -159,6 +165,22 @@ class SupplierApplicationForm extends React.Component {
...
+          readOnly={false /* TODO: only supplier creator can edit his supplier info */}
+          actionUrl={this.context.simPublicUrl}
+          supplierId={userInfo.supplierid}
+          locale={userInfo.locale}
+          username={userInfo.id}
+          onChange={this.handleDirtyState}
+        />
+      </I18nBundle>
+    );
+
```
Comment (gr4per) [inline src/client/components/SupplierApplicationForm/SupplierApplicationForm.js]: not clear why we need to define I18n, should be transparent here as handled between component and app.i18n

File: src/client/components/SupplierApplicationForm/SupplierApplicationForm.js (hunk 1 above)
Comment (lasong) [inline src/client/components/SupplierApplicationForm/SupplierApplicationForm.js]: I didn't have any strong reason for doing it this way. Will be changed in a separate PR

File: src/client/components/SupplierApplicationForm/SupplierApplicationForm.js (hunk 2 above)
Comment (PunkSage) [inline src/client/components/SupplierApplicationForm/SupplierApplicationForm.js]: Well, I think that this formatting is coherent in what is defined in ESLint currently. Maybe we should autofix the current code or change .eslintrc to match our style guide?
